documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,932 @@
+# Migrate from v1 to v2 of the AWS IoT SDK for JavaScript
+
+The AWS IoT SDK for JavaScript v2 is a major rewrite of the AWS IoT Device SDK for JavaScript v1 code base. It includes many updates, such as improved
+consistency, ease of use, more detailed information about client status. This guide describes the major features that
+are new in the v2 SDK, and provides guidance on how to migrate your code to v2 from v1 of the AWS IoT SDK for JavaScript.
+
+> [!NOTE]
+> If you can't find the information you need in this guide, visit the [How to get help](#how-to-get-help) section for more help and guidance.
+
+* [What's new in AWS IoT Device SDK for JavaScript v2](#whats-new-in-aws-iot-device-sdk-for-javascript-v2)
+* [How to get started with AWS IoT Device SDK for JavaScript v2](#how-to-get-started-with-aws-iot-device-sdk-for-javascript-v2)
+    * [Package name change](#package-name-change)
+    * [MQTT protocol](#mqtt-protocol)
+    * [Browser applications](#browser-applications)
+    * [Client builder](#client-builder)
+    * [Client start](#client-start)
+    * [Connection types and features](#connection-types-and-features)
+    * [Lifecycle events](#lifecycle-events)
+    * [Publish](#publish)
+    * [Subscribe](#subscribe)
+    * [Unsubscribe](#unsubscribe)
+    * [Client stop](#client-stop)
+    * [Client shutdown](#client-shutdown)
+    * [Reconnects](#reconnects)
+    * [Offline operations queue](#offline-operations-queue)
+    * [Logging](#logging)
+    * [Client for AWS IoT Device Shadow](#client-for-aws-iot-device-shadow)
+    * [Client for AWS IoT Jobs](#client-for-aws-iot-jobs)
+    * [Client for AWS IoT fleet provisioning](#client-for-aws-iot-fleet-provisioning)
+    * [Example](#example)
+* [How to get help](#how-to-get-help)
+* [Appendix](#appendix)
+    * [MQTT5 features](#mqtt5-features)
+
+> [!NOTE]
+> There are multiple v2 references in the v1 SDK documentation. These references does **not** refer to the v2 SDK, but to the release versions of the v1 SDK after v2.0.0.
+
+## What's new in AWS IoT Device SDK for JavaScript v2
```

Comment:
I'm unfamiliar with the V1 SDK but it was called out specifically in the java migration guide that v2 was async while v1 was not. Is that not a difference in JS? Did v1 also support browser and non-browser iot?

Reply:
Reading further below, it appears as though v1 can be used in a browser through use of browserify and webpack. If that's at all a hassle to do or tricky to use, might be worth calling out that v2 doesn't have those requirements? Bret or Vera would know better than me whether this is a significant enough thing to be listed here.